### PR TITLE
Fix screen flickering in SDL2

### LINKED ISSUE #6422: Screen flickering on tablet outside viewable area
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Other

### Describe the bug

There is annoying screen flickering on Android tablet when the game is in 4:3 screen mode (any resolution). 
OS: Android 8.1.0
Device: Lenovo TB-X304L
Native screen resolution: 1280x800
Steps to reproduce:
1. Set game resolution to 640x480 or 800x600. 
2. Tap anywhere on the screen.
Result: everytime when the screen is being tapped there is flickering outside image borders (in the space where normally should be blank screen because image is 4:3 and the screen itself is 16:10).

### Save file

Not necessary - it can be reproduced even in main menu.

### Additional info

https://user-images.githubusercontent.com/10282847/209815240-abc9c24a-740f-40f8-9293-aa9af78422a9.mp4

https://user-images.githubusercontent.com/10282847/209815017-487b8536-12ef-4476-af97-839619668f16.mp4

### PR DESCRIPTION
**Pull request by ihhub**
close #6422

### CODE REVIEW

File: src/engine/screen.cpp [hunk 1]
```diff
@@ -1205,7 +1205,12 @@ namespace
                 return false;
             }
 
-            int returnCode = SDL_SetRenderTarget( _renderer, nullptr );
+            int returnCode = SDL_SetRenderDrawColor( _renderer, 0, 0, 0, SDL_ALPHA_OPAQUE );
+            if ( returnCode < 0 ) {
+                ERROR_LOG( "Failed to set default color for rendered. The error value: " << returnCode << ", description: " << SDL_GetError() )
```
Comment (oleg-derevenetz) [inline src/engine/screen.cpp:1210]: Hi @ihhub it seems that there is a typo: `default color for rendered` -> `default color for renderer` (?)

Comment (ihhub) [reply src/engine/screen.cpp]: Yup, it was a typo.

**Comment by oleg-derevenetz** [comment]
Hi @ihhub I noticed that you don't call `SDL_RenderClear()` when you don't update a full frame. Could this be related to this issue?

**Comment by ihhub** [comment]
> Hi @ihhub I noticed that you don't call `SDL_RenderClear()` when you don't update a full frame. Could this be related to this issue?

Yes. `SDL_RenderClear()` should be used for every call. Somehow on other systems is not needed but looks like for Android (only for this specific case) is.

**Comment by ihhub** [comment]
@oleg-derevenetz , during Adventure Map we render a full frame so not exactly related to this.

**Comment by oleg-derevenetz** [comment]
@ihhub I have an impression that flickering occurs when we move the software cursor (which does not require a full frame rendering by itself) AND we don't do the full frame rendering at the same time. Take s look on the second video - there is no flickering when adventure map is scrolled.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**